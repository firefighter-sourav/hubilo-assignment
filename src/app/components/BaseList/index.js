import PropTypes from 'prop-types'
import React from 'react'

import { _posts } from 'app/store/selectors/posts'

import * as Styles from './styles'

import BaseItem from '../BaseItem'
/**
 * Generic List for all of similar types list view
 * @param {object} props
 * @returns List of items
 */
const BaseList = (props) => {
  const { data } = props

  return (
    <Styles.Wrapper>
      {data.map((item) => (
        <BaseItem key={item.id} item={item} />
      ))}
    </Styles.Wrapper>
  )
}

BaseList.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
}

export default BaseList
