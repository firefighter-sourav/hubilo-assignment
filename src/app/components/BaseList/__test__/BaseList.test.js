import {
  cleanup,
  render as testRender,
  fireEvent,
} from '@testing-library/react'
import React from 'react'
import ReactDOM from 'react-dom'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

import BaseList from '..'

afterEach(cleanup)

const data = [
  {
    id: 1,
    title: 'Hello',
    body: {
      id: 1,
      content: 'There',
    },
  },
  {
    id: 2,
    title: 'World',
    body: {
      id: 1,
      content: 'God is almighty',
    },
  },
]

it('renders ConfigSelect without crasing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<BaseList data={data} />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('matches snapshot', () => {
  const tree = renderer.create(<BaseList data={data} />).toJSON()
  expect(tree).toMatchSnapshot()
})
