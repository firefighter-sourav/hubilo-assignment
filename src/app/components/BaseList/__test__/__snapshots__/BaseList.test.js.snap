// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bdnxRM eaWVJw"
>
  <div
    className="sc-gtsrHT ckIFxw"
  >
    <div
      className="sc-dlnjwi dHpSnY"
    >
      Hello
    </div>
    <div
      className="sc-hKFxyN cqFfPS"
    >
      id
      : 
      1
    </div>
    <div
      className="sc-hKFxyN cqFfPS"
    >
      content
      : 
      There
    </div>
  </div>
  <div
    className="sc-gtsrHT ckIFxw"
  >
    <div
      className="sc-dlnjwi dHpSnY"
    >
      World
    </div>
    <div
      className="sc-hKFxyN cqFfPS"
    >
      id
      : 
      1
    </div>
    <div
      className="sc-hKFxyN cqFfPS"
    >
      content
      : 
      God is almighty
    </div>
  </div>
</div>
`;
